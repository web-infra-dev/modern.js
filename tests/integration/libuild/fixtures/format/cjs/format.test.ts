import { getLibuilderTest } from '@/toolkit';

describe('fixture:format:cjs', () => {
  it('format cjs', async () => {
    const bundler = await getLibuilderTest({
      root: __dirname,
      format: 'cjs',
    });
    await bundler.build();

    bundler.expectJSOutputMatchSnapshot();

    const bundler2 = await getLibuilderTest({
      root: __dirname,
      format: 'cjs',
      sourceMap: true,
    });
    await bundler2.build();

    bundler.expectJSOutputMatchSnapshot();
  });
});
