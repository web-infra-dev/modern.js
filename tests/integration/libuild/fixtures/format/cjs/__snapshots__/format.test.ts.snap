// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixture:format:cjs format cjs 1`] = `
"var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === \\"object\\" || typeof from === \\"function\\") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, \\"__esModule\\", { value: true }), mod);

// index.ts
var cjs_exports = {};
__export(cjs_exports, {
  a: () => a
});
module.exports = __toCommonJS(cjs_exports);
function a() {
  console.log(1);
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  a
});
"
`;

exports[`fixture:format:cjs format cjs 2`] = `
"var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === \\"object\\" || typeof from === \\"function\\") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, \\"__esModule\\", { value: true }), mod);

// index.ts
var cjs_exports = {};
__export(cjs_exports, {
  a: () => a
});
module.exports = __toCommonJS(cjs_exports);
function a() {
  console.log(1);
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  a
});
"
`;
